rs/ic_os/config/src/types.rs

```diff
@@ -83,12 +84,13 @@ pub struct BackupSpoolSettings {
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
 pub struct ICOSSettings {
+    pub mgmt_mac: FormattedMacAddress,
```

Comment:
mgmt_mac is now a non-optional parameter in the config. 
If a mgmt_mac is supplied via deployment.json, we will use this one. Else, the mgmt_mac is determined in the config tool initial call to CreateSetuposConfig

This change will allow us to later delete the repeat code in setupos and hostos_tool dynamically checking for a config mgmt_mac and if not found, using get_ipmi_mac to generate it.

Additionally, it allows us to consolidate the setup-hostname scripts because now that mgmt_mac is in the config object, guestos will be able to determine it's hostname the same way hostos does, and it won't need to be passed in a hostname from hostos.